connectors/sources/sharepoint_online.py

```diff
@@ -946,6 +946,8 @@ def get_default_configuration(cls):
         }
 
     async def validate_config(self):
+        self.configuration.check_valid()
```

Comment:
```suggestion
        super().validate_config()
```

I think this pattern can also apply to other sources that are using `self.configuration.check_valid()`

Reply:
Yeah but for now I chose the consistent call here - same as in other connectors.

TBH I also think that `configuration.check_valid()` should be called outside - I think all connectors need to have valid configuration to function. If we follow current pattern, some connectors can by mistake or on purpose ignore it and allow invalid config, which is just weird. I think we should revisit this.

Reply:
What do you think if we merge it like it is now and make a tech debt issue to address this?

Reply:
++ on making a tech debt issue

Reply:
Created an issue: https://github.com/elastic/connectors-python/issues/1194